pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -785,13 +796,13 @@ func TestMonitorNodeHealth(t *testing.T) {
 
 	for testName, tt := range tests {
 		t.Run(testName, func(t *testing.T) {
-			_, ctx := ktesting.NewTestContext(t)
+			tCtx := ktesting.Init(t)
```

Comment:
should we initialize the context outside the loop? Like you do in newly added `TestProcessPodMarkPodNotReady`?

Reply:
The closer a context is to the actual test code the better, i.e. innermost. That way the output is associated with `testName` in this case, not with the parent test.

Reply:
Thanks. @carlory please change the new method `TestProcessPodMarkPodNotReady` then to initialize context inside the loop

Reply:
updated.